### PR TITLE
fix(ColorMode): Fixing bug with usePrefersDarkMode

### PR DESCRIPTION
**Pull request by thedoomshine**
### Overview

Fixing bug causing errors in datadog

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [X] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/30649)  | [Monolith Env](https://pr-30649-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/1931)  | [Portal Env](https://tengu.codecademy.com/article/accessibility-and-html?PR_ENV=portal-app-pr-1931)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.2.2-alpha.76dfa1.0<br/>@codecademy/gamut-illustrations@0.26.2-alpha.76dfa1.0<br/>@codecademy/gamut-kit@0.6.122-alpha.76dfa1.0<br/>@codecademy/gamut-labs@31.11.11-alpha.76dfa1.0<br/>@codecademy/gamut-patterns@0.8.5-alpha.76dfa1.0<br/>@codecademy/gamut-styles@13.1.1-alpha.76dfa1.0<br/>@codecademy/gamut-tests@4.0.4-alpha.76dfa1.0<br/>@codecademy/gamut@47.1.2-alpha.76dfa1.0<br/>@codecademy/markdown-overrides@0.9.2-alpha.76dfa1.0<br/>@codecademy/monaco@0.6.41-alpha.76dfa1.0<br/>@codecademy/styleguide@57.0.11-alpha.76dfa1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6279662f4f0148007a740ec6--gamut-preview.netlify.app>https://6279662f4f0148007a740ec6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6279662f4f0148007a740ec6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.2.2-alpha.bf324c.0<br/>@codecademy/gamut-illustrations@0.27.1-alpha.bf324c.0<br/>@codecademy/gamut-kit@0.6.124-alpha.bf324c.0<br/>@codecademy/gamut-labs@31.11.13-alpha.bf324c.0<br/>@codecademy/gamut-patterns@0.8.5-alpha.bf324c.0<br/>@codecademy/gamut-styles@13.1.1-alpha.bf324c.0<br/>@codecademy/gamut-tests@4.0.4-alpha.bf324c.0<br/>@codecademy/gamut@47.2.2-alpha.bf324c.0<br/>@codecademy/markdown-overrides@0.9.2-alpha.bf324c.0<br/>@codecademy/monaco@0.6.43-alpha.bf324c.0<br/>@codecademy/styleguide@57.1.2-alpha.bf324c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627a9078b5a65b46ced4e52d--gamut-preview.netlify.app>https://627a9078b5a65b46ced4e52d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627a9078b5a65b46ced4e52d>Deploy Logs</a></p>

File: packages/gamut-styles/src/ColorMode.tsx [hunk 1]
```diff
@@ -83,13 +83,17 @@ export function usePrefersDarkMode() {
...
       setPrefersDarkMode(event.matches);
     }
 
-    mq.addEventListener('change', onChange);
+    if (mq && 'addEventListener' in mq) {
+      setPrefersDarkMode(mq.matches);
+      mq.addEventListener('change', onChange);
+    }
+
+    return () => mq.removeEventListener('change', onChange);
```
Comment (MattSchiller) [inline packages/gamut-styles/src/ColorMode.tsx:96]: Shouldn't this be inside the `if` block? So if we never add the event listener, we don't then also try to remove it?

Comment (thedoomshine) [reply packages/gamut-styles/src/ColorMode.tsx:96]: yep! good catch

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.2.2-alpha.5f2293.0<br/>@codecademy/gamut-illustrations@0.27.1-alpha.5f2293.0<br/>@codecademy/gamut-kit@0.6.125-alpha.5f2293.0<br/>@codecademy/gamut-labs@31.11.14-alpha.5f2293.0<br/>@codecademy/gamut-patterns@0.8.5-alpha.5f2293.0<br/>@codecademy/gamut-styles@13.1.1-alpha.5f2293.0<br/>@codecademy/gamut-tests@4.0.4-alpha.5f2293.0<br/>@codecademy/gamut@47.3.1-alpha.5f2293.0<br/>@codecademy/markdown-overrides@0.9.2-alpha.5f2293.0<br/>@codecademy/monaco@0.6.44-alpha.5f2293.0<br/>@codecademy/styleguide@57.2.1-alpha.5f2293.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627d4ce6b4a00e00563853bf--gamut-preview.netlify.app>https://627d4ce6b4a00e00563853bf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627d4ce6b4a00e00563853bf>Deploy Logs</a></p>

**Review by aresnik11 — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.3.1-alpha.997b7e.0<br/>@codecademy/gamut-illustrations@0.27.1-alpha.997b7e.0<br/>@codecademy/gamut-kit@0.6.127-alpha.997b7e.0<br/>@codecademy/gamut-labs@31.12.2-alpha.997b7e.0<br/>@codecademy/gamut-patterns@0.8.5-alpha.997b7e.0<br/>@codecademy/gamut-styles@13.1.1-alpha.997b7e.0<br/>@codecademy/gamut-tests@4.0.4-alpha.997b7e.0<br/>@codecademy/gamut@47.3.2-alpha.997b7e.0<br/>@codecademy/markdown-overrides@0.9.2-alpha.997b7e.0<br/>@codecademy/monaco@0.6.45-alpha.997b7e.0<br/>@codecademy/styleguide@57.4.1-alpha.997b7e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62828636b93663009eb11cc2--gamut-preview.netlify.app>https://62828636b93663009eb11cc2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62828636b93663009eb11cc2>Deploy Logs</a></p>

**Pull request merged**